### PR TITLE
JBPM-9214 The zoom does not work when start a new process from Proces…

### PR DESCRIPTION
**Pull request by bxf12315**
…s Definition

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@bxf12315 have you checked if the issue could be caused by the wrong element being selected on https://github.com/kiegroup/jbpm-wb/blob/master/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramWidgetViewImpl.java#L108 
Iḿ wonering that it could be two dics with same id ( processDiagramDiv ) at the point. Maybe you just need to have different ids?

**Comment by bxf12315** [comment]
> @bxf12315 have you checked if the issue could be caused by the wrong element being selected on [/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramWidgetViewImpl.java@`master`#L108](https://github.com/kiegroup/jbpm-wb/blob/master/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramWidgetViewImpl.java?rgh-link-date=2020-07-07T13%3A25%3A25Z#L108)
> Iḿ wonering that it could be two dics with same id ( processDiagramDiv ) at the point. Maybe you just need to have different ids?

Yes you are right, the reason is two same id div in on page, Proces Definiton detail have ProcessDiagramPresenter, Process instance detail also have the component. if we use different id, we need create new ProcessDefinitionDiragram for it and it total same to ProcessDiagramPresenter. I think ProcessDiagramPresenter is reuseable for process defintions and process instance .

**Comment by cristianonicolai** [comment]
@bxf12315 just create a new unique id every time `ProcessDiagramWidgetViewImpl` is created? then use that as selector.

**Comment by bxf12315** [comment]
xswa

> @bxf12315 just create a new unique id every time `ProcessDiagramWidgetViewImpl` is created? then use that as selector.

we inject it in here https://github.com/kiegroup/jbpm-wb/blob/7e60a5274b01b790d2ef313961b061ca89672de1/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramWidgetViewImpl.java#L48.
https://github.com/kiegroup/jbpm-wb/blob/523f3596e47896f24ec8e822b04d16a8c1412d33/jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramWidgetViewImpl.html#L9

I'm not sure how to refactor them. The PR just want to keep mini change.

**Comment by bxf12315** [comment]
@cristianonicolai Follow your comment, I use change id way and it's work.
But in the solution, one page has two same functions components, we just change one div id of them, there also have many same id's div(same functions). I'm not sure what problems this will cause， but it's work and does not have an issue.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public void setProcessImageService(final Caller<ProcessImageService> processImag
     public void onProcessSelectionEvent(@Observes final ProcessDefSelectionEvent event) {
         view.showBusyIndicator(constants.Loading());
         view.disableExpandAnchor();
+        view.setProcessDiagramDivId("ProcessDefDiagram");
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java:49]: the presenter should not be responsible for defining an id in the view. Simply generated a ramdom one on the view side?
`DOM.createUniqueId();`

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java:49]: if use the way, one page two "<div class="col-md-12" data-field="diagramContainerDiv" data-i18n-prefix="ProcessDiagramWidgetViewImpl.">", is it reasonable?

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java]: updated, Please review again.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/definition/list/ProcessDefinitionListPresenter.java [hunk 2]
```diff
@@ -57,12 +57,14 @@
...
 import org.uberfire.workbench.model.menu.Menus;
 
+import static org.jbpm.workbench.common.client.PerspectiveIds.PROCESS_DEFINITION_DETAILS_SCREEN;
 import static org.jbpm.workbench.common.client.PerspectiveIds.SEARCH_PARAMETER_PROCESS_DEFINITION_ID;
 import static org.jbpm.workbench.common.client.util.DataSetUtils.getColumnStringValue;
 import static org.jbpm.workbench.pr.model.ProcessDefinitionDataSetConstants.*;
 import static org.kie.workbench.common.services.shared.resources.PerspectiveIds.PROCESS_INSTANCES;
 @Dependent
 @WorkbenchScreen(identifier = PerspectiveIds.PROCESS_DEFINITION_LIST_SCREEN)
+@SuppressWarnings("PMD.TooManyStaticImports")
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/definition/list/ProcessDefinitionListPresenter.java:67]: I dont think any of these changes are needed to fix the issue. could you revert it please.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java (hunk 1 above)
Comment (cristianonicolai) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java]: sorry but I didnt get whats the issue here.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/definition/list/ProcessDefinitionListPresenter.java (hunk 2 above)
Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/definition/list/ProcessDefinitionListPresenter.java]: done

**Comment by jstastny-cz** [comment]
Why haven't been me or @prashuts asked for review here? This has broken our tests.

**Comment by bxf12315** [comment]
@jstastny-cz sorry I forget it, which test broken?

**Comment by prashuts** [comment]
@bxf12315  - Tests related to ProcessInstanceDiagram where script mainly involves Process Diagram panels. Scripts failed due to change in Locator attriubute id value (older - processDiagramDiv)

**Comment by bxf12315** [comment]
Yes According to the @cristianonicolai's comments, use createUniqueId to generate a id of the div.
So Can I do some help for fix the issue?

**Pull request merged**